enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -5980,6 +5980,456 @@ class AdjointGenerator
+          arg2 =
+              BuilderZ.CreateBitCast(malins, call.getArgOperand(9)->getType());
+#if LLVM_VERSION_MAJOR >= 10
+          BuilderZ.CreateMemCpy(
+              arg2, MaybeAlign(),
+              gutils->getNewFromOriginal(call.getArgOperand(9)), MaybeAlign(),
+              Bsize);
+#else
+          BuilderZ.CreateMemCpy(
+              arg2, 0, gutils->getNewFromOriginal(call.getArgOperand(9)), 0,
+              Bsize);
+#endif
+        }
+        if (Acache && Bcache) {
+          auto valins1 = BuilderZ.CreateInsertValue(undefinit, arg1, 0);
+          cacheval = BuilderZ.CreateInsertValue(valins1, arg2, 1);
+        } else if (Acache)
+          cacheval = arg1;
+        else if (Bcache)
+          cacheval = arg2;
+        gutils->cacheForReverse(BuilderZ, cacheval,
+                                getIndex(&call, CacheType::Tape));
+      }
+      if (Mode == DerivativeMode::ReverseModeCombined ||
+          Mode == DerivativeMode::ReverseModeGradient) {
+        IRBuilder<> Builder2(call.getParent());
+        getReverseBuilder(Builder2);
+        auto dfunc = gutils->oldFunc->getParent()->getOrInsertFunction(
+            funcName, Builder2.getVoidTy(), Builder2.getInt32Ty(),
+            Builder2.getInt32Ty(), Builder2.getInt32Ty(), Builder2.getInt32Ty(),
+            Builder2.getInt32Ty(), Builder2.getInt32Ty(),
+            call.getArgOperand(6)->getType(), call.getArgOperand(7)->getType(),
+            Builder2.getInt32Ty(), call.getArgOperand(7)->getType(),
+            Builder2.getInt32Ty(), call.getArgOperand(6)->getType(),
+            call.getArgOperand(7)->getType(), Builder2.getInt32Ty());
+        auto oneval = Builder2.getInt32(1);
+        auto doneval = Builder2.CreateSIToFP(oneval, innerType);
+        Value *sabtrans, *saldb, *sbatrans, *sblda, *saldc, *salda, *sbldb,
+            *sbldc;
+        Value *structarg1, *structarg2;
+        if (Acache || Bcache) {
+          if (Mode == DerivativeMode::ReverseModeGradient &&
+              (aactive || bactive)) {
+            cacheval = BuilderZ.CreatePHI(cachetype, 0);
+          }
+          cacheval =
+              lookup(gutils->cacheForReverse(BuilderZ, cacheval,
+                                             getIndex(&call, CacheType::Tape)),
+                     Builder2);
+          if (Acache && Bcache) {
+            structarg1 = BuilderZ.CreateExtractValue(cacheval, 0);
+            structarg2 = BuilderZ.CreateExtractValue(cacheval, 1);
+          } else if (Acache)
+            structarg1 = cacheval;
+          else if (Bcache)
+            structarg2 = cacheval;
+        }
+        if (!Acache)
+          structarg1 = lookup(gutils->getNewFromOriginal(call.getArgOperand(7)),
+                              Builder2);
+        if (!Bcache)
+          structarg2 = lookup(gutils->getNewFromOriginal(call.getArgOperand(9)),
+                              Builder2);
+        if (cast<ConstantInt>(call.getArgOperand(0))->getValue() == 102) {
```

Comment:
Can you compare with a character rather than a raw number? Also usually don't both upper and lowercase work?

Reply:
The first three arguments are enums. I am not sure about chars that we can pass as arguments instead. Do you have an example?

Do you instead mean passing in the enum names like `CblasRowMajor`, `CblasNoTrans`, etc instead of the number?